release/src/main/scripts/build_release_candidate.sh

```diff
@@ -96,20 +102,42 @@ if [[ $confirmation = "y" ]]; then
   git checkout ${RELEASE_BRANCH}
   RELEASE_COMMIT=$(git rev-parse --verify ${RELEASE_BRANCH})
 
-  echo "-------------Building Java Artifacts with Gradle-------------"
-  git config credential.helper store
+  echo "[Current Step]: Build and stage java artifacts"
+  echo "Do you want to proceed (skip if retrying staging step)? [y|N]"
+  read confirmation
+  if [[ $confirmation = "y" ]]; then
+    echo "-------------Building Java Artifacts with Gradle-------------"
+    git config credential.helper store
+
+    ./gradlew release -Prelease.newVersion=${RELEASE}-SNAPSHOT \
```

Comment:
Can we just not do this?

    set_release_version --release "${RELEASE}"
    git commit -m "Build ${RC_TAG}"
    git tag "${RC_TAG}"
    git push --tags
    git push

Then we don't have random commits that are immediately reverted.

Reply:
Hey Kenn, could you please explain more about why the situation you described above could happen?

Reply:
I don't believe we currently use the reverted commit in the release process anyway. My guess would be that the aim was to use it to generate javadoc (without SNAPSHOT), possibly build other stuff. I don't know what reverting it is supposed to accomplish. Perhaps not to break the script that generates the next RC?

Reply:
Exactly. It has to do with stripping `-SNAPSHOT` from Java and `.dev` from Python. I don't know of another purpose. As far as I can tell, it is left over from CVS & SVN, where branches are expensive and/or confusing. The maven release plugin is simulating things that are no longer required with git, leaving a confusing commit history.

Here it is, in bad drawings...

With maven or gradle release plugin:

    x.y.z-SNAPSHOT ----> x.y.z (tag x.y.z-rc1) --revert-->  x.y.z-SNAPSHOT (branch release-x.y.z)

With basic git usage:

    x.y.z-SNAPSHOT (branch release-x.y.z)
       \
        x.y.z (tag x.y.z-rc1)

Related: https://lists.apache.org/thread.html/205472bdaf3c2c5876533750d417c19b0d1078131a3dc04916082ce8%40%3Cdev.beam.apache.org%3E

Reply:
A couple things to note:

 - no reverting commits on the release branch (because why add noise?)
 - branch is always SNAPSHOT
 - rc is always a tag (a dead end) signaling you cannot put more commits on top of it (I confirmed that any tag is not GC'd byt GitHub)

It might be clever to do this:

1. set version and push RC tag
2. have totally automated process build RC from the tag